WalletWasabi/Crypto/ZeroKnowledge/Prover.cs

```diff
@@ -0,0 +1,98 @@
+using NBitcoin.Secp256k1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Crypto.Randomness;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public static class Prover
```

Comment:
what's the plan for this class with respect to compound proofs, such as conjunctions etc? if the pattern of a prover being a factory for various kinds of proofs will remain, building them in a modular way would probably require two phases:

1. have a factory for *statement* objects (not proofs, but intents to prove which contain all of the inputs necessary to make a proof) which can then be combined by AND/OR operations into compound statements
2. have the prover finalize a compound statement into a concrete proof transcript

i think a one shot API can still be provided for the standard compound proofs, but the more basic building blocks such as knowledge of DL or repr would need that kind of splitting to be reusable

Reply:
Maximize decisions not made. The plan is that if it grows too big or too complex (in other words unmaintainable) then CodeScene will complain and we have to change things around.

Reply:
this was not a question about size but about the intended API

Reply:
to clarify the question: if the elementary proof types are meant to be generic, i.e. first you construct some lower level proofs, then you combine them into a larger one, then the lower level API needs to have two phases, one to prepare the statements, and then another to have the prover actually generate the transcript, which would not be possible with the current API that does everything in one go from arguments to finalized proof object